x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningExtension.java

```diff
@@ -25,6 +25,10 @@ default void configure(Settings settings) {}
 
     boolean isNlpEnabled();
 
+    default boolean isLearningToRankEnabled() {
+        return false;
```

Comment:
Why do we need a default here? I think it would be better to rely on the MachineLearningExtension impl to do the right thing?

Reply:
If there isn't a default here when the PR is first merged then it will be impossible for another team member to easily update the submodule in the serverless repo. It will only be possible to update the submodule as part of the change that implements this method in the serverless ML extension.

This is why so many changes that span the public and serverless repos have to be spread over 3 PRs.

The sequence could be:

1. Merge this as-is
2. Update the submodule in serverless
3. Merge the serverless PR to override this method in the serverless extension
4. Merge another public PR to remove the default

Reply:
I see, thanks for explaining

Reply:
Indeed @droberts195 is right. I added the default to keep ensure serverless will continue to work when updating the submodule.

When the submodule will be updated, we will have to implement it in serverless implementation of the extension and remove the default implementation in the interface.